tests/third_party/cupy/math_tests/test_matmul.py

```diff
@@ -161,31 +177,73 @@ class TestMatmulLarge(unittest.TestCase):
     }
 
     @testing.for_all_dtypes(name="dtype1")
-    @testing.numpy_cupy_allclose(rtol=1e-3, atol=1e-3)  # required for uint8
-    def test_operator_matmul(self, xp, dtype1):
+    @testing.for_all_dtypes(name="dtype2")
+    @testing.numpy_cupy_allclose(
+        rtol=1e-3, atol=1e-3, type_check=has_support_aspect64()
+    )  # required for uint8
+    def test_operator_matmul(self, xp, dtype1, dtype2):
         if (dtype1, dtype1) in self.skip_dtypes or (
             dtype1,
             dtype1,
         ) in self.skip_dtypes:
             return xp.array([])
         x1 = testing.shaped_random(self.shape_pair[0], xp, dtype1)
-        x2 = testing.shaped_random(self.shape_pair[1], xp, dtype1)
+        x2 = testing.shaped_random(self.shape_pair[1], xp, dtype2)
         return operator.matmul(x1, x2)
 
     @testing.for_all_dtypes(name="dtype1")
-    @testing.numpy_cupy_allclose(rtol=1e-3, atol=1e-3)  # required for uint8
-    def test_cupy_matmul(self, xp, dtype1):
+    @testing.for_all_dtypes(name="dtype2")
+    @testing.numpy_cupy_allclose(
+        rtol=1e-3, atol=1e-3, type_check=has_support_aspect64()
+    )  # required for uint8
+    def test_cupy_matmul(self, xp, dtype1, dtype2):
         if (dtype1, dtype1) in self.skip_dtypes or (
             dtype1,
             dtype1,
         ) in self.skip_dtypes:
             return xp.array([])
         shape1, shape2 = self.shape_pair
         x1 = testing.shaped_random(shape1, xp, dtype1)
-        x2 = testing.shaped_random(shape2, xp, dtype1)
+        x2 = testing.shaped_random(shape2, xp, dtype2)
+        return xp.matmul(x1, x2)
+
+
+@pytest.mark.parametrize(
+    "shape1, shape2",
+    [
+        # TODO: include it when issue #1540 in dpctl is resolved
+        # ((256, 256, 3, 2), (256, 256, 2, 4)),
+        ((256, 256, 3, 2), (2, 4)),
+        ((3, 2), (256, 256, 2, 4)),
+    ],
+)
+class TestMatmulIntegralLargeBatch:
+    @testing.for_int_dtypes(name="dtype")
+    @testing.numpy_cupy_array_equal()
+    def test_operator_matmul(self, xp, dtype, shape1, shape2):
+        x1 = testing.shaped_random(shape1, xp, dtype)
+        x2 = testing.shaped_random(shape2, xp, dtype)
+        return operator.matmul(x1, x2)
+
+    @testing.for_int_dtypes(name="dtype")
+    @testing.numpy_cupy_array_equal()
+    def test_cupy_matmul(self, xp, dtype, shape1, shape2):
+        x1 = testing.shaped_random(shape1, xp, dtype)
+        x2 = testing.shaped_random(shape2, xp, dtype)
         return xp.matmul(x1, x2)
 
 
+@pytest.mark.skip("until issue #1540 in dpctl is resolved")
```

Comment:
I guess the issue will be closed due to classified as undefined behavior, so probably need to update skip reason.